util/chplspell

```diff
@@ -0,0 +1,271 @@
+
+# Complication: We want to invoke scspell once for all the .chpl,
+# .cpp, .h, README* files (etc.), and again with --no-c-escapes for
+# all the .tex files.
+
+# This is all a hack.  Longer term, the work done by this script should
+# be mostly done by scspell, so this won't be needed.  In the meantime,
+# - Identify directories to be spell checked, so we can generate the
+#   flat file list that scspell wants.
+# - Certain scspell options require us to take care not to reorder
+#   their args (--merge-file-ids and --rename-file).
+# - If any files or directories are specified, spell check only those.
+# - Otherwise, check all the $DEFAULTFILES, and the globbed files under
+#   the $DEFAULTDIRS.
+
+while true; do
+    case "$1" in
+	--help)
+	    $RIVENV $SCSPELL --help
+	    echo;
+	    echo "chplspell sets --override-dictionary to $CHPLDICT"
+	    echo "and --relative-to to $CHPL_HOME"
+	    echo "and passes the file contents of named directories to scspell"
+	    echo "Use -- to have chplspell pass the remainder of its args"
+	    echo "unchanged."
+	    exit 0
+	    ;;
+
+	# Pass-through args with no parameter handled by -*) case below
+
+	# No support for these through chplspell
+	--override-dictionary|-set-dictionary|--relative-to)
+	    echo "No support for $1 via chplspell." >&2
+	    exit 1
+	    ;;
+
+	# Pass-through args with two parameters.  Pass via $ARGS to preserve
+	# ordering and not attempt directory expansion
+	--merge-file-ids|--rename-file)
+	    ARGS+=("$1" "$2" "$3")
+	    shift 3
+	    JUSTARGS=1
+	    ;;
+	--delete-files)
+	    DELETE_INVOKED=1
+	    JUSTARGS=1
+	    ARGS+=("--delete-files")
+	    shift
+	    ;;
+
+	--) shift
+	    # Remaining args remain in "$@", and "$#" will be non-zero.
+	    break
+	    ;;
+	-*) ARGS+=("$1"); shift;;  # just pass through e.g. --report-only
+	"") break;;
+	*)
+	    if [ $DELETE_INVOKED == 1 ]; then
+		# delete goes through JUSTARGS case at end, so we need
+		# the files to be deleted to be in ARGS.
+		ARGS+=("$1")
+	    elif [ -f "$1" ]; then
+		# User is responsible for specifying --no-c-escapes
+		# as needed when listing regular files; we'll pick it up
+		# via ARGS.
+		NORMALFILES+=("$1")
+	    elif [ -d "$1" ]; then
+		TARGETDIRS+=("$1")
+	    else
+		# Non-existing files, etc, pass through.  Especially as
+		# scspell requires a "filename" even for e.g. -i.
+		ARGS+=("$1")
+	    fi
+	    shift
+	    ;;
+	esac
+done
+
+if [ $JUSTARGS == 0 ]; then
+    # If nothing was specified on the command line, use our default lists.
+    if [ ${#TARGETDIRS} == 0 -a ${#NORMALFILES} == 0 -a $# == 0 ]; then
+	TARGETDIRS=("${DEFAULTDIRS[@]}")
+	NORMALFILES=("${DEFAULTFILES[@]}")
+	cd $CHPL_HOME
+    fi
+
+    for d in "${TARGETDIRS[@]}"; do
+	# globstar enables the patterns above (e.g. **/*.chpl) to recurse.
+	# nullglob causes a glob that matches nothing to return an empty
+	# string instead of the literal glob.
+	shopt -s globstar nullglob
```

Comment:
Whoops. The `#!/bin/bash` was invoking my system bash, while the upgraded bash was installed from homebrew, resulting in confusion.

My system bash is: `GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin15)`

Note that `$BASH` and `$BASH_VERSION` point to the newer version. Consequently, changing the shebang to`#!/usr/bin/env bash` resolves the issue.